docs/sources/flow/reference/components/discovery.file.md

```diff
@@ -0,0 +1,124 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/discovery.file
+title: discovery.file
+---
+
+# discovery.file
+
+`discovery.file` discovers files on the local filesystem using the [doublestar][] library.
+
+[doublestar]: https://github.com/bmatcuk/doublestar
+
+## Usage
+
+```river
+discovery.file "LABEL" {
+  paths = ["DOUBLESTAR_PATH"]
```

Comment:
Let's fix this, since now it doesn't expect just the glob, but a list of targets

Couple of more points
a) Do you think we should we rename the Argument to `targets`?
b) Do you think we should expliticly call out that the component looks out for the `__path__` key in the input targets?
c) Seems like we're not using the `exclude_paths` Argument anymore, but getting them from the labels, right?

Reply:
Good call on b and c. As far as A I could go either way, paths feel more intuitive to me, but targets are more type correct.

Reply:
I'd recommend we go with 'targets' so it can more easily be read (eg. using the output of `kubernetes.discovery.default.targets`)

Reply:
IMO after thinking on it for 5, I think it should not be targets just in case we allow referencing arguments at some point in river.

Reply:
After conversation went with path_targets